zilencer/management/commands/populate_db.py

```diff
@@ -1174,25 +1185,27 @@ def bulk_create_reactions(all_messages: List[Message]) -> None:
     Reaction.objects.bulk_create(reactions)
 
 
-def choose_date_sent(num_messages: int, tot_messages: int, threads: int) -> datetime:
+def choose_date_sent(
+    num_messages: int, tot_messages: int, starting_n_days_ago: int, threads: int
+) -> datetime:
     # Spoofing time not supported with threading
     if threads != 1:
         return timezone_now()
 
-    # Distrubutes 80% of messages starting from 5 days ago, over a period
-    # of 3 days. Then, distributes remaining messages over past 24 hours.
+    # Distrubutes 80% of messages starting from starting_n_days_ago days ago, over a period
+    # of the first 60% of those days. Then, distributes remaining messages over past 24 hours.
```

Comment:
I wonder if we just want min(`N-2`, 1) rather than 60% here?  I think the intent of this algorithm was just to ensure (1) some messages in the last 4 hours, (2) some >24hr gaps between adjacent messages, and (3) a decent bulk of messages in the last day so you see adjacent messages with the same date too, and that seems like a cleaner way to translate that idea.

Reply:
Ah, this is great context of what functionality you need here! I'll make those changes.